### PR TITLE
Upgrade Adapter for Lusas v21.0

### LINKED ISSUE #377: Upgrade to Lusas v21
#### Description:

Due to be released on November 27th 2023: 
[Lusas21 Microsite](https://www.lusas.com/?utm_source=https%3a%2f%2fenews.lusas.com%2flusaslz%2f&utm_medium=email&utm_campaign=LUSAS+Version+21%3a+Global+release+date+announcement&utm_term=LUSAS+Version+21%3a+Global+release+date+announced&utm_content=5902&gator_td=x30RZsPJYhDy31Yq5crDNnDwmFfBntFfSPZwP1AyoaWg91TLbsVs4EPduztGn%2bIZDZlxAyw2BNhMetxJMmwHiIHT0gOLZUOFMuuCLMvJSaXez5tSU%2bcSNpNg5tfBpktUX3XZb%2fQZnrrF3AOjI21LxB1l7iOgqKZrzNvSYMD2F%2bOnq9u8chL3Dj9DbjQEb8DQ)

### PR DESCRIPTION
**Pull request by KalleEdstroem**
### Issues addressed by this PR

Closes #377 

### Test files

[Push and pull elements, properties and tags](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FIssue%2FBHoM%2FLusas%5FToolkit%2F%23377%20Upgrade%20Adapter%20for%20Lusas%2021%2E0%2FBuroHappold%5FBHoM%5Fv6%2E1%2Ebeta%5FCreate%20%2D%20read%20%2D%20elements%20%2D%20bridges%2Egh&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&parent=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FIssue%2FBHoM%2FLusas%5FToolkit%2F%23377%20Upgrade%20Adapter%20for%20Lusas%2021%2E0 )

[Push and Pull Loads](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FIssue%2FBHoM%2FLusas%5FToolkit%2F%23377%20Upgrade%20Adapter%20for%20Lusas%2021%2E0%2FBuroHappold%5FBHoM%5Fv6%2E1%2Ebeta%5FCreate%20%2D%20read%20%2D%20elements%20%2D%20bridges%2Egh&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&parent=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FIssue%2FBHoM%2FLusas%5FToolkit%2F%23377%20Upgrade%20Adapter%20for%20Lusas%2021%2E0)

### Changelog

Added adapter functionality for Lusas 21.0

### Additional comments

Tested with the testing scripts used for Beta releases.
- Tested with Lusas 20.0 Adapter and New Lusas 21.0 Adapter and compared behavior to be identical. 
- Checked so that Pushed and pulled had the same properties and that it matched what the software showed.

Testing will require you to have multiple versions of Lusas installed.

### CODE REVIEW

File: Lusas_Engine/Lusas_Engine.csproj [hunk 1]
```diff
@@ -107,6 +107,24 @@
...
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release210|AnyCPU'">
+    <OutputPath>bin\Release210\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <Optimize>true</Optimize>
+    <DebugType>pdbonly</DebugType>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <LangVersion>7.3</LangVersion>
+    <ErrorReport>prompt</ErrorReport>
+  </PropertyGroup>
```
Comment (peterjamesnugent) [inline Lusas_Engine/Lusas_Engine.csproj:127]: Is this needed? Pretty sure the build configs are only needed for the Adapter (that references the Lusas LPI).

File: Lusas_oM/Lusas_oM.csproj [hunk 2]
```diff
@@ -123,6 +123,24 @@
...
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release210|AnyCPU'">
+    <OutputPath>bin\Release210\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <Optimize>true</Optimize>
+    <DebugType>pdbonly</DebugType>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <LangVersion>7.3</LangVersion>
+    <ErrorReport>prompt</ErrorReport>
+  </PropertyGroup>
```
Comment (peterjamesnugent) [inline Lusas_oM/Lusas_oM.csproj:143]: As above, oM should not have different build configs.

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Have tested with the following scripts:
BuroHappold_BHoM_v7.0.beta_Create read loads
BuroHappold_BHoM_v7.0.beta_Create - read - elements - bridges
BuroHappold_BHoM_v7.0.beta_ExecuteCommands
BuroHappold_BHoM_v7.0.beta_Pull Structural Analysis Results

The branch needs to be rebased to capture the bug fixes at the tail end of 7.0 milestone.

The oM and Engine do not need alternate build configs.

You will need to include alternate configs here:
https://github.com/BHoM/BHoM_Installer/blob/main/IncludedRepos/altConfigs.txt

**Comment by peterjamesnugent** [comment]
Can't see anything here: https://www.lusas.com/releases/v21.0/21.0-0/LUSAS%2021.0%20Error%20Fix%20and%20Modification%20Release%20Note.pdf that would affect the toolkit.

File: Lusas_Engine/Lusas_Engine.csproj (hunk 1 above)
Comment (KalleEdstroem) [reply Lusas_Engine/Lusas_Engine.csproj:127]: This follows from that all other build configs is mentioned above and that It was added in the update for version 20.0.

**Review by peterjamesnugent — DISMISSED** [review]
Have retested with the force push changes - raised a separate issue here #392 (unrelated to the PR) - happy to approve.

**Comment by peterjamesnugent** [comment]
@BHoMBot check required
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by KalleEdstroem** [comment]
@BHoMBot check core

**Comment by KalleEdstroem** [comment]
@BHoMBot check required
@BHoMBot check copyright-compliance

**Comment by KalleEdstroem** [comment]
@BHoMBot check core
@BHoMBot project-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check project-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check null-handling
@BHoMBot check serialisation

**Review by peterjamesnugent — APPROVED** [review]
Approved with the build configs reinstated for oM and Engine.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**